### PR TITLE
Enh ps stuff

### PR DESCRIPTION
**Pull request by danielballan**
- Test subs input
- Always represent collections of subs / sub_factories as dicts.
- Perform normalization from flexible input at set using traitlets where possible, otherwise using a property.

### CODE REVIEW

**Comment by danielballan** [comment]
Heads up, @klauer, this is an API break for `default_subs` and `default_sub_factories`: they are dicts, not lists. This improves performance for calbacks that don't need to see Events.

Change your code from `[cb]` to `{'all': [cb]}`.

**Comment by danielballan** [comment]
P.S. As a user convenience, we normalize input at set time:

`ascan.subs = cb` -> `ascan.subs = {'all': cb}`

but users will have to know about the dict if they try to append to existing collections.

File: bluesky/simple_scans.py [hunk 1]
```diff
@@ -61,14 +62,17 @@ def plot_motor(scan):
     return LivePlot(gs.PLOT_Y, scan.motor._name)
 
 
-class _BundledScan:
-    default_subs = []  # subscriptions
-    default_sub_factories = []  # functions that take a scan, return a sub
+class _BundledScan(HasTraits):
```
Comment (tacaswell) [inline bluesky/simple_scans.py]: Do not need `HasTraits` here anymore?

Comment (danielballan) [inline bluesky/simple_scans.py]: Fixed and pushed

**Pull request merged**